### ISSUE TITLE
Poor error with invalid CPU environment group

### ISSUE THREAD

**Issue by AustinSchuh** [bazelbuild/bazel#3911]
I have a BUILD file in //tools with the following in it.

```
environment(name = "k8")
environment(name = "armhf-debian")
environment(name = "roborio")
environment(name = "cortex-m4f")
environment_group(
    name = "cpus",
    environments = [":k8", ":roborio", ":armhf-debian", ":cortex-m4"],
    defaults = [":k8", ":roborio", ":armhf-debian"],
)
```

(Notice the missing 'f' in the environments list).

I get a pretty cryptic error

```
$ bazel test -c opt //... --experimental_auto_cpu_environment_group=//tools:cpus
ERROR: package contains errors: tools.
ERROR: error loading package 'tools': Package 'tools' contains errors.
INFO: Elapsed time: 0.414s
ERROR: Couldn't start the build. Unable to run tests.
```

I would expect to see something saying that 'cortex-m4' isn't a valid target, or something so I don't have to guess and check.

**Comment by gregestren** [comment]
This is an obscure error coming out of the intersection of environment_group, package error reporting, and [Skyframe](https://bazel.build/designs/skyframe.html).

The *actual* error, as you know, is that `environment_group` `cpus` references a non-existent target: `:cortex-m4`.

Normally errors like this get faithfully reported here: https://github.com/bazelbuild/bazel/blob/a729b9b4c3d7844a7d44934bf3365f92633c0a60/src/main/java/com/google/devtools/build/lib/skyframe/PackageFunction.java#L632-L635

In plain English, this is essentially saying: "*in the process of loading a package, when we've **pretty much** finished initializing the package through its Builder, check the Builder for errors and report them*".

The problem is with _**pretty much**_. There's still a small finalizer step that happens afterwards: https://github.com/bazelbuild/bazel/blob/a729b9b4c3d7844a7d44934bf3365f92633c0a60/src/main/java/com/google/devtools/build/lib/skyframe/PackageFunction.java#L640

This is mostly uninteresting cleanup before returning the `Package` instance. But it includes this crucial step:
https://github.com/bazelbuild/bazel/blob/a1f90ce376a82a862ffb7180e39f8f13e64488f6/src/main/java/com/google/devtools/build/lib/packages/Package.java#L1404

which checks *exactly* the error you're triggering in your above example.

This has to happen at the end of package loading to guarantee all targets that *do* exist have already been loaded, so we don't incorrectly omit a legitimate target that just wasn't visited yet. But because this happens *after* the error reporting happened, the error gets swallowed.

It should be sufficient to move the error reporting step after package building. I'm testing just that change to see if it's generally compatible with Bazel.

This is what the error looks like with that change

``` 
ERROR: /home/workspace/tools/BUILD:5:1: environment
//tools:cortex-m4 does not exist.
ERROR: error loading package 'tools': Package 'tools' contains errors.
```
